### DISCUSSION TITLE
Moving vite.config.ts on different location

### DISCUSSION THREAD

**Discussion by vizo** [ElMassimo/vite_ruby#245] [Q&A] [answered]
Hey..  If i move **vite.config.ts** from Rails root to "app/frontend" (because it's not loaded), I get error: 
Building with Vite ⚡️
vite v3.0.9 building for development...
✓ 0 modules transformed.
Could not resolve entry module (app/frontend/index.html).
error during build:
Error: Could not resolve entry module (app/frontend/index.html).

If i run by:
`bin/vite build app/frontend --clear --mode=development`
works as expected ...

I think commands bin/vite serve and build should respect config root directory from "/config/vite.json"

Is there any other way to make it work out of the box?

Thank you in advance.

**Comment by ElMassimo** [comment, marked as answer]
Vite allows specifying a different config file, by passing [`--config`](https://vitejs.dev/config/#configuring-vite):

```sh
bin/vite dev -- --config app/frontend/vite.config.ts
bin/vite build -- --config app/frontend/vite.config.ts
```

I would recommend keeping `vite.config.ts` at the root, since that's the default behavior for any Vite project. 

By using a different configuration it's likely that you'll run into specific problems most users don't run into.

---

> should respect config root directory from "/config/vite.json"

The root has higher priority than `config/vite.json` ([which is resolved relative to the root](https://vite-ruby.netlify.app/config/index.html#configpath)), so that happens the other way around.
